sympy/holonomic/holonomic.py

```diff
@@ -2647,17 +2649,16 @@ def _extend_y0(Holonomic, n):
                     r = r.as_expr()
                 sol += a * r
             y1.append(sol)
-            list_red = _derivate_diff_eq(list_red)
+            list_red = _derivate_diff_eq(list_red, K)
 
         return y0 + y1[len(y0):]
 
 
-def DMFdiff(frac):
+def DMFdiff(frac, K):
     # differentiate a DMF object represented as p/q
     if not isinstance(frac, DMF):
         return frac.diff()
 
-    K = frac.ring
     p = K.numer(frac)
```

Comment:
We cannot get the ring from the `DMF` directly. Instead the ring must be passed explicitly if it is needed.